app/services/post_service.py

```diff
@@ -0,0 +1,31 @@
+from typing import Any
+import uuid
+from app.models import Post
+from app.utils.file import get_image_url_dir, FORM_POST_IMAGE
+from datetime import datetime
+from app.utils.file import get_file_extension, get_image_url
+
+
+def create_post(post: Post, file: Any) -> None:
+    Post.insert(post)
+
+    upload_post_image(file, post.image_url)
+
+
+def upload_post_image(file: Any, image_url: str) -> None:
+    file.save(get_image_url_dir(image_url))
+
+
+def post_builder(request: Any, user_id: int) -> Post:
```

Comment:
Could be just moved to `create_post()` as a 1 liner once the create post Flask form is implemented. Similar to `auth_service.sign_up()`.